onnxruntime/core/providers/cpu/tensor/unsqueeze.cc

```diff
@@ -43,14 +43,15 @@ Status UnsqueezeBase::PrepareCompute(OpKernelContext* ctx, Prepare& p) const {
 
   std::vector<int64_t> axes;
   size_t num_inputs = ctx->InputCount();
-  if (num_inputs == 2) { //axes is an input
+  if (num_inputs == 2) {  //axes is an input
     const Tensor* axes_tensor = ctx->Input<Tensor>(1);
     ORT_ENFORCE(axes_tensor != nullptr, "Axes input is null");
-    ORT_ENFORCE(axes_tensor->Shape().NumDimensions() == 1,
-                "An axes tensor must be a vector tensor.");
-    auto nDims = static_cast<size_t>(axes_tensor->Shape()[0]);
+    ORT_ENFORCE(axes_tensor->Shape().NumDimensions() == 0 ||
```

Comment:
A looser check is fine - the loop test uses a scalar to specify the axes for an unsqueeze node in its subgraph and the spec doesn't say that it is an error for a non 1-D case.